### PR TITLE
Ddennis

### PR DESCRIPTION
**Pull request by dewaynedennis**
I had to change/update a lot of files to get the initializing of the model to work so @thread-speaker, @dabok, @swlyons if you guys want to check and make sure that this is all kosher...thanks.

### CODE REVIEW

File: java/src/client/managers/GameManager.java [hunk 1]
```diff
@@ -1,16 +1,37 @@
...
 import java.util.Date;
-import shared.definitions.ResourceType;
+        
 import shared.definitions.DevCardType;
+import shared.definitions.ResourceType;
+import shared.locations.VertexLocation;
+import shared.locations.EdgeLocation;
+import shared.locations.HexLocation;
 
+<<<<<<< HEAD
```
Comment (thread-speaker) [inline java/src/client/managers/GameManager.java:21]: Looks like there might still be a little bit of a conflict in GameManager

File: java/src/client/managers/GameManager.java [hunk 2]
```diff
@@ -243,4 +300,17 @@ public boolean useCard(int gameID, Card c) {
     public void endTurn(int gameID) {
 
     }
+<<<<<<< HEAD
```
Comment (thread-speaker) [inline java/src/client/managers/GameManager.java:303]: This is a remnant of your merge conflict earlier.

File: java/src/client/managers/UserManager.java [hunk 3]
```diff
@@ -18,8 +18,13 @@ public UserManager(){
      * @pre
      * @post returns if the user was successfully created or not
      */
+<<<<<<< HEAD
```
Comment (thread-speaker) [inline java/src/client/managers/UserManager.java:21]: Not really sure how UserManager got merge conflicts. It doesn't look like you changed anything...

**Comment by thread-speaker** [comment]
This looks good, except for the lingering merge conflict artifacts. Still not sure why UserManager has them...

**Comment by swlyons** [comment]
Yeah in several of the files there are  >>>>>>>
7071a9788250be0deba93f8906181d9d67b2a583     and   <<<<<<< HEAD  and  ======
=. Getting rid of duplicate files in the data package and keeping them in
the shared package is fine (I wanted to swap them but it's fine as
is). Also you got rid of equivalent which is bad for location manager and
map manager. The method equivalent checks to see if the content is the
same, the equals method, on the other hand, checks to see if they're the
same object. Most locations that .equivalent got replaced with .equals will
always return false.

File: java/src/client/managers/UserManager.java (hunk 3 above)
Comment (dewaynedennis) [inline java/src/client/managers/UserManager.java:21]: I didn't think I did either

**Comment by dewaynedennis** [comment]
Not sure why, but there wasn't an equivalent method in my location manager or map manager so I changed them so I could compile. I'll change 'em back in my next commit if you still wanna go that route (with the equivalent method instead of the equals method)

**Comment by swlyons** [comment]
The methods aren't in location manager or map manager. They were in the
files that were duplicated in the shared package (EdgeLocation,
HexLocation, VertexLocation). If I wanna go that route implies that there's
an alternative. What is your suggestion?

**Comment by dewaynedennis** [comment]
I'm sure you've already thought about it, but I was thinking maybe we could overwrite the equals method so that it would check the content being equal as well

**Comment by swlyons** [comment]
Let me provide a larger explanation so we are all on the same track.

The reason I didn't overwrite the equals method was in case it was being
used elsewhere in the project. The way they had it was "unusual" to say the
least.
 Perhaps we'll want to test if they're the same object regardless if they
have the same content in the future. We don't want to test if they're the
same object in locationManager and mapManager. We just want to test if
their content is the same, this is because I know that they are not the
same object. For instance.

VertexLocation unusableLocation2 = new VertexLocation(new HexLocation
(location.getNormalizedLocation().getHexLoc().getX() + 1, location.
getNormalizedLocation().getHexLoc().getY()), VertexDirection.NorthWest);
VertexLocation unusableLocation3 = new VertexLocation(new HexLocation
(location.getNormalizedLocation().getHexLoc().getX() + 1, location.
getNormalizedLocation().getHexLoc().getY() - 1), VertexDirection.NorthWest);
 for (Location locationUnavailable : unsettledLocations) {
  if (locationUnavailable.getNormalizedLocation().equivalent
(unusableLocation1)
 || locationUnavailable.getNormalizedLocation().equivalent
(unusableLocation2)
 || locationUnavailable.getNormalizedLocation().equivalent(unusableLocation3))
{
 locationUnavailable.setCanBeSettled(false);
 }
 }

When a settlement is built, a settlement cannot be built in the neighboring
locations.
This part of the method handles when a settlement is built, set the
neighboring VertexLocation's ability to be settled to false.
unusableLocation2 and unusableLocatio3 are new objects. Therefore they are
not the same as the neighboring locations. However they are in the same
spot (equivalent).
The for-each loop goes through all locations until it finds the locations
that are equivalent to those just created and sets them to false since the
Locations in unsettledLocations is the actual permanent data for the game.
This method is not the most efficient in the world, however, I couldn't
come up with one more efficient.
If we had an equals method and an equivalent method, I don't guarantee we
will use the equals method elsewhere. I think it'd be good to keep just in
case we need to test if two things are the same object regardless of their
content.
What are your thoughts? Does this make sense?

**Comment by dewaynedennis** [comment]
Yeah I see what you're saying

**Pull request closed without merging**